### DISCUSSION TITLE
Keycloak-nodejs-connect deprecation is there any other alternatives ?

### DISCUSSION THREAD

**Discussion by vishwasmehraZestl** [keycloak/keycloak#23551] [Q&A]
At my organisation we want to use full capabilities of keycloak as it is suitable to our usecase and want to intergrate in our nest js backend, but recently found out that this package has been deprecated. Since it was server side package that supported server side nodejs apps what are the other better alternatives ? I came across 'Keycloak-js' library from the keycloak docs but it says it's for client side, still can we somehow use it in for our backend process ? Any help would be appreciated. Thanks !!

**Comment by jonkoops** [comment]
> I came across 'Keycloak-js' library from the keycloak docs but it says it's for client side, still can we somehow use it in for our backend process ?

No, it is exclusively a client-side library and cannot be used in a Node.js environment. I'd recommend reading the [existing discussion](https://github.com/keycloak/keycloak/discussions/9818) around this subject.
